samples/stm32l4/basic-sample/src/iotconnect_app.c

```diff
@@ -291,28 +339,119 @@ bool app_startup(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr) {
 #ifdef IOTCONNECT_SYMETRIC_KEY
     config->auth.type = IOTC_KEY;
     config->auth.data.symmetric_key = IOTCONNECT_SYMETRIC_KEY;
+#else
+    config->auth.type = IOTC_X509;
+
+#ifdef ENABLE_DDIM_TO_DRIVER_SAMPLE
+    struct to_driver_parameters parameters = {0};
+    parameters.operational_identity_slot = 2;
+    parameters.operational_csr_slot = 2;
+    parameters.bootstrap_identity_slot = 1;
+    IotcDdimInterface ddim_interface;
+    IotcAuthInterfaceContext auth_context;
+    if(to_create_auth_driver( //
+            &(config->auth.data.x509.auth_interface), //
+            &ddim_interface, //
+			&auth_context, //
+			&parameters)) { //
+    	return false;
+    }
+    config->auth.data.x509.auth_interface_context = auth_context;
+    uint8_t serial[TO_HW_SN_SIZE];
+    size_t serial_size;
+    if(config->auth.data.x509.auth_interface.get_serial(
+    		auth_context, //
+			serial, //
+			&serial_size//
+			)) {
+    	return false;
+    }
+    printf("DDIM:   SN: ");
+	for (int i=0; i < serial_size; i++) {
+		printf("%02X", serial[i]);
+	}
+	printf("\r\n");
+
+#ifdef DDIM_TEST
+    int failed = 0;
+    for (int test_num = 1; test_num < 100; test_num++) {
+    	if(iotcdi_obtain_operational_identity(&azrtos_config, &ddim_interface, config->auth.data.x509.auth_interface_context)) {
+    		printf("Test #%d failed\r\n", test_num);
+    		failed++;
+    	} else {
+    		printf("Test #%d success\r\n", test_num);
+    	}
+        tx_thread_sleep(30 * NX_IP_PERIODIC_RATE);
+    }
+    printf("Failed %d tests\r\n", failed);
+    return false;
+
+#endif
```

Comment:
Should this be an #else ?
Minor issue -- just saw cppcheck complain about unreachable code after a return.

Reply:
Removed that code completely.